cmd/cniplugins/passt-binding/pkg/plugin/plugin.go

```diff
@@ -101,11 +100,16 @@ func (c *cmd) CmdAddResult(args *skel.CmdArgs) (types.Result, error) {
 		netname := netConf.Args.Cni.LogicNetworkName
 		log.Printf("setup for logical network %s completed successfully", netname)
 
-		podLink, lerr := c.netlinkAdapter.ReadLink(primaryPodInterfaceName)
-		if lerr != nil {
-			return lerr
+		defaultGatwayLinks, err := link.DiscoverByDefaultGateway(vishnetlink.FAMILY_ALL)
+		if err != nil {
+			return err
+		}
+
+		if len(defaultGatwayLinks) != 1 {
+			return fmt.Errorf("unexpected number of default gw links")
 		}
 
+		podLink := defaultGatwayLinks[0]
```

Comment:
same here, it seems that it"ll pick the first matched link, please consider simplifying `DiscoverByDefaultGateway` to return the first matched link.